### PR TITLE
Construct Amount without CompressedCommitment.

### PR DESCRIPTION
**Pull request by samdealy**
### Motivation
Fog no longer stores a compressed commitment in a TxOutRecord, and the Android SDK's OwnedTxOut constructor uses this compressed commitment to construct an Amount object. As such, we need a new way to create an Amount object without a compressed commitment.

### In this PR
This PR creates a new Amount constructor that doesn't use a compressed commitment. This constructor calls into the native "android-bindings" libs, so this PR contains the necessary Java native interface methods and contains the compiled "android-bindings" libs. 

Fixes [ANDROID-87](https://mobilecoin.atlassian.net/browse/ANDROID-87?atlOrigin=eyJpIjoiMTMyYmI5ZjhhODFhNDEwMmEzYjg4NmVmNTE1NjA4NWIiLCJwIjoiaiJ9).

### Future Work
Update the Fog repo's android-bindings directory, which contains the Rust code that compiles into the .so libs included in this PR.

On the Fog side, [PR 99](https://github.com/mobilecoinfoundation/fog/pull/99) adds the Rust methods that implement that native method interfaces.

### CODE REVIEW

**Review by voloshyn — COMMENTED** [review]
Please link android-bindings PR for reference

**Comment by samdealy** [comment]
> Please link android-bindings PR for reference

Done.

File: android-sdk/src/main/java/com/mobilecoin/lib/Amount.java [hunk 1]
```diff
@@ -43,6 +43,28 @@
         }
     }
 
+    /**
+     * Constructs native Amount object from the txOutSharedSecret and masked value.
+     *
+     * @param txOutSharedSecret  A {@link RistrettoPrivate} representing the shared secret.
```
Comment (voloshyn) [inline android-sdk/src/main/java/com/mobilecoin/lib/Amount.java:49]: this has to be RistrettoPublic
```suggestion
     * @param txOutSharedSecret  A {@link RistrettoPublic} representing the shared secret.
```

**Review by voloshyn — APPROVED** [review]
LGTM

File: android-sdk/src/main/java/com/mobilecoin/lib/Amount.java (hunk 1 above)
Comment (samdealy) [reply android-sdk/src/main/java/com/mobilecoin/lib/Amount.java]: Done.

**Pull request merged**